pkg/app/consume.go

```diff
@@ -0,0 +1,135 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/prometheus-pulsar-remote-write/pkg/pulsar"
+	"github.com/grafana/prometheus-pulsar-remote-write/pkg/remote"
+)
+
+type consumeCommand struct {
+	app  *App
+	name string
+
+	sendTimeout        time.Duration
+	pulsar             *pulsarConfig
+	pulsarSubscription string
+	remoteWriteURL     string
+	batchMaxDelay      time.Duration
+}
+
+func newConsumeCommand(app *App) *consumeCommand {
+	name := "consume"
+	c := &consumeCommand{
+		name:   name,
+		app:    app,
+		pulsar: &pulsarConfig{},
+	}
+
+	cmd := app.app.Command(name, "Consume metrics on the pulsar bus and send them as remote_write requests")
+
+	c.pulsar.addFlags(cmd)
+	cmd.Flag("send-timeout", "The timeout to use when sending samples to the remote_write endpoint.").
+		Default("30s").DurationVar(&c.sendTimeout)
+	cmd.Flag("pulsar.subscription", "The subscription name used to consume messages of the bus. It is important that if you are reading with multiple consumers, all of them need to share the same subscription name.").
+		Default("pulsar-adapter").StringVar(&c.pulsarSubscription)
+	cmd.Flag("remote-write.url", "The URL of remote_write endpoint to forward the metrics to.").Required().
+		StringVar(&c.remoteWriteURL)
+
+	return c
+}
+
+func (c *consumeCommand) pulsarClient() (*pulsar.Client, error) {
+	// TODO: Not too sure how relevant it is for consuming from the bus
+	//clientOptions.OperationTimeout = p.readTimeout
```

Comment:
is this comment still relevant? I can't see the vars `clientOptions` nor `p`, so it looks like maybe that line would be outdated now anyway?

Reply:
I think it's still relevant, something to ask when I am speaking to a pulsar expert next:  6b31f27364e23b541796745d4c25234906c8bca3